lib/arc/file.ex

```diff
@@ -15,7 +15,10 @@ defmodule Arc.File do
   # Given a remote file
   def new(remote_path = "http" <> _) do
     case save_file(remote_path) do
```

Comment:
I'd prefer to use the same parsing logic for determining the file name and extension (and only do it once).

Can you change this to:

```elixir
def new(remote_path = "http" <> _) do
  uri = URI.parse(remote_path)
  filename = Path.basename(uri.path)

  case save_file(uri, filename) do
    {:ok, local_path} -> %Arc.File{path: local_path, file_name: filename}
    :error -> {:error, :invalid_file_path}
  end
end

defp save_file(uri, filename) do
  local_path =
    generate_temporary_path()
    |> Kernel.<>(Path.extname(filename))

  # rest of this method...
end
```